self_test/generic/complex_matrices_test/cr_complex_matrix_test.cc

```diff
@@ -0,0 +1,249 @@
+      cout << matrix(i, j) << ", ";
+    }
+    cout << endl;
+  }
+}
+
+int main()
+{
+  // test default constructor
+  CRComplexMatrix matrix_default;
+  cout << matrix_default.nrow() << endl;
+  cout << matrix_default.ncol() << endl;
+
+  constexpr unsigned long n_col_rect = 2;
+  constexpr unsigned long n_row_rect = 4;
+  // Number of non-zero entries
+  constexpr unsigned n_nz_rect = 4;
+
+  Vector<complex<double>> value_rect(n_nz_rect);
+  value_rect[0] = complex<double>(1.0, 1.0);
+  value_rect[1] = complex<double>(2.0, 1.0);
+  value_rect[2] = complex<double>(3.0, 1.0);
+  value_rect[3] = complex<double>(4.0, 1.0);
+
+  Vector<int> column_index_rect(n_nz_rect);
+  column_index_rect[0] = 0;
+  column_index_rect[1] = 2;
+  column_index_rect[2] = 1;
+  column_index_rect[3] = 3;
+
+  // Final entry stores a fictitous index equal to the number of non-zeros
+  Vector<int> row_start_rect(n_col_rect + 1);
+  row_start_rect[0] = 0;
+  row_start_rect[1] = 2;
+  row_start_rect[2] = n_nz_rect;
+
+  // test full matrix constructor with a rectangular matrix
+  CRComplexMatrix matrix_rect(
+    value_rect, column_index_rect, row_start_rect, n_col_rect, n_row_rect);
+  cout << matrix_rect.nrow() << endl;
+  cout << matrix_rect.ncol() << endl;
+  print_cr_complex_matrix(matrix_rect);
+
+  constexpr unsigned long n_col_square = 4;
+  constexpr unsigned long n_row_square = 4;
+  constexpr unsigned n_nz_square = 7;
+
+  Vector<complex<double>> values_square(n_nz_square);
+  values_square[0] = complex<double>(1.0, 1.0);
+  values_square[1] = complex<double>(2.0, -1.0);
+  values_square[2] = complex<double>(3.0, 0.1);
+  values_square[3] = complex<double>(4.0, 3.0);
+  values_square[4] = complex<double>(5.0, 1.0);
+  values_square[5] = complex<double>(6.0, -2.0);
+  values_square[6] = complex<double>(0.5, -1.0);
+
+  Vector<int> column_index_square(n_nz_square);
+  column_index_square[0] = 0;
+  column_index_square[1] = 2;
+  column_index_square[2] = 0;
+  column_index_square[3] = 1;
+  column_index_square[4] = 1;
+  column_index_square[5] = 2;
+  column_index_square[6] = 3;
+
+  // Final entry stores a fictitous index equal to the number of non-zeros
+  Vector<int> row_start_square(n_col_square + 1);
+  row_start_square[0] = 0;
+  row_start_square[1] = 2;
+  row_start_square[2] = 4;
+  row_start_square[3] = 6;
+  row_start_square[4] = n_nz_square;
+
+  // test full matrix constructor with a square matrix
+  CRComplexMatrix matrix_square(values_square,
+                                column_index_square,
+                                row_start_square,
+                                n_col_square,
+                                n_row_square);
+  cout << matrix_square.nrow() << endl;
+  cout << matrix_square.ncol() << endl;
+  print_cr_complex_matrix(matrix_square);
+
+  // test LU decomposition
+  cout << matrix_square.ludecompose() << endl;
+
+  constexpr unsigned long vector_length = n_row_square;
+
+  Vector<std::complex<double>> rhs(vector_length);
```

Comment:
Inconsistent use of `std::` to clarify scope of `complex<double>`. I recommend removing it